components/script/dom/htmlelement.rs

```diff
@@ -345,7 +345,7 @@ impl HTMLElement {
             NodeTypeId::Element(ElementTypeId::HTMLElement(type_id)) =>
                 match type_id {
                     HTMLElementTypeId::HTMLInputElement =>
-                        self.downcast::<HTMLInputElement>().unwrap().Type() != "hidden",
+                        &*self.downcast::<HTMLInputElement>().unwrap().type_() != "hidden",
```

Comment:
The entire point of atoms is avoiding character-by-character string comparisons, so yes.